src/IR/LambdaLift.hs

```diff
@@ -85,17 +93,26 @@ addCurrentScope :: I.VarId -> LiftFn ()
 addCurrentScope v =
   modify $ \st -> st { currentScope = S.insert v $ currentScope st }
 
--- | Get a fresh variable name
-getFresh :: LiftFn String
+-- | Create a string composed of the scope trail and a variable name.
+prependTrail :: Identifier -> LiftFn Identifier
+prependTrail cur = do
+  curTrail <- gets trail
+  return $ foldr (<>)
```

Comment:
Replace `foldr .. ..` with `mconcat` (from `Data.Monoid`)

Reply:
You can also replace `Identifier "_"` with just `"_"`; `OverloadedStrings` inserts a `fromString` for you

Reply:
This entire few lines can be reduced to  `return $ mconcat $ intersperse "_" $ reverse $ cur : curTrail`